core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -500,7 +515,7 @@ public static SqlCall stripSeparator(SqlCall call) {
    * Returns the element type of a MULTISET.
    */
   public static final SqlReturnTypeInference MULTISET_ELEMENT_NULLABLE =
-      MULTISET.andThen(SqlTypeTransforms.TO_MULTISET_ELEMENT_TYPE);
+      MULTISET.andThen(SqlTypeTransforms.TO_COLLECTION_ELEMENT_TYPE);
```

Comment:
Do you think SEQ or LIST better than COLLECTION?

Reply:
@JiajunBernoulli i modify the name because it can be use not only multiset type, but also array type

Reply:
`COLLECTION` usually also includes `MAP` and `SET`, does them also work?

Reply:
@JiajunBernoulli COLLECTION includes array and multiset, both have getComponentType, but doesn't include map type. map type has getKeyType and getValueType